src/plugins/data_source_management/public/components/constants.tsx

```diff
@@ -17,3 +18,29 @@ export const NO_DATASOURCES_CONNECTED_MESSAGE = 'No data sources connected yet.'
 export const CONNECT_DATASOURCES_MESSAGE = 'Connect your data sources to get started.';
 export const NO_COMPATIBLE_DATASOURCES_MESSAGE = 'No compatible data sources are available.';
 export const ADD_COMPATIBLE_DATASOURCES_MESSAGE = 'Add a compatible data source.';
+
+export const OPENSEARCH_DOCUMENTATION_URL =
+  'https://opensearch.org/docs/latest/dashboards/management/data-sources/';
+
+export const OPENSEARCH_S3_DOCUMENTATION_URL =
+  'https://opensearch.org/docs/latest/dashboards/management/S3-data-source/';
+
+export const OPENSEARCH_ACC_DOCUMENTATION_URL =
+  'https://opensearch.org/docs/latest/dashboards/management/accelerate-external-data/';
+export const QUERY_RESTRICTED = 'query-restricted';
+export const QUERY_ALL = 'query-all';
+
+export const DatasourceTypeToDisplayName: { [key in DatasourceType]: string } = {
+  PROMETHEUS: 'Prometheus',
+  S3GLUE: 'Amazon S3',
+};
+
+export const PrometheusURL = 'Prometheus';
+export const AmazonS3URL = 'AmazonS3AWSGlue';
+
+export const UrlToDatasourceType: { [key: string]: DatasourceType } = {
+  [PrometheusURL]: 'PROMETHEUS',
+  [AmazonS3URL]: 'S3GLUE',
+};
+
+export type AuthMethod = 'noauth' | 'basicauth' | 'awssigv4';
```

Comment:
not seeing this being used yet. but is this to be used later when integrating with the existing auth work done by MDS? https://github.com/opensearch-project/OpenSearch-Dashboards/blob/cf4e496aa03d12b18244f271185863f5fbf1ff6d/src/plugins/data_source/public/plugin.ts#L27.

if not, we should get a design proposal, as this might be doing a rehash of heavy lifting done by that work which also has saved objects integration.